docker-images/kafka/scripts/zookeeper_run.sh

```diff
@@ -51,5 +53,34 @@ fi
 
 . ./set_kafka_gc_options.sh
 
+############ ZK Upgrade Start ############
+# ZK 3.4.x -> 3.5.x upgrade adds snapshot checks that can cause ZK pods with persistent storage to fail on startup
+# The section below disables the snapshot checks until the servers have created a snapshot.
+
+# If this is a 3.4.x server then place an indicator file in the data directory so this persists over multiple restarts
+flag_file=${ZOOKEEPER_DATA_DIR}/zk_upgrade_from_3_4
+if [[ $(ls libs | grep -Po 'zookeeper-\K3.4.\d+' | head -1) ]]; then
+    touch ${flag_file}
+fi
+
+# If this server is attached to a disk that was once used by a 3.4.x server then we may need to disable snapshot checks
+if [[ -f ${flag_file} ]]; then
+    # If the version-2 folder does not exist then this is first start up of a 3.4.x server therefore this is not an upgrade so we can skip
+    if [[ -d ${ZOOKEEPER_DATA_DIR}/version-2 ]]; then
+        # If the 3.4 flag is there and the data directory is present this might be an upgrade so check for snapshots
+        if [[ $(find ${ZOOKEEPER_DATA_DIR}/version-2 -maxdepth 1 -name "snapshot.*") ]]; then
+            # If there are snapshot files then we will have a stable server after the upgrade and can remove the flag file
+            rm ${flag_file}
+        else
+            # If there are no snapshot files and we started from a 3.4.x server then disable the snapshot checks for this boot
+            KAFKA_OPTS="$KAFKA_OPTS -Dzookeeper.snapshot.trust.empty=true"
```

Comment:
You should pass it through the configuration file to make it clear which configuration we are using - using the env var and system properties kind of hides it from logs.

Reply:
After much digging it seems putting this setting in the config file means it is never used. It is only used if passed as a system property. Should I add some echo lines here so they show up in the log?

Reply:
Or do you want me to add it to the config even though it is not used so it shows up in the log?

Reply:
Well, if it is not used you should not add it there. But some note about this in the log would be useful. So you should log it IMHO.